adapter/oam.go

```diff
@@ -167,16 +167,23 @@ func (or *OAMRegistrant) Register() error {
 	return nil
 }
 
+type MeshModelConfig struct {
+	Category    string
+	SubCategory string
+	Metadata    map[string]interface{}
+}
+
 // StaticCompConfig is used to configure CreateComponents
 type StaticCompConfig struct {
-	MeshModelName string //Used in Adding ModelName onto Core Meshmodel components. Pass it the same as meshName in OAM components
-	URL           string //URL
-	Method        string //Use the constants exported by package. Manifests or Helm
-	OAMPath       string //Where to store the directory.(Each directory will have an array of definitions and schemas)
-	MeshModelPath string
-	DirName       string           //The directory's name. By convention, it should be the version name
-	Config        manifests.Config //Filters required to create definition and schema
-	Force         bool             //When set to true, if the file with same name already exists, they will be overridden
+	MeshModelName   string //Used in Adding ModelName onto Core Meshmodel components. Pass it the same as meshName in OAM components
+	URL             string //URL
+	Method          string //Use the constants exported by package. Manifests or Helm
+	OAMPath         string //Where to store the directory.(Each directory will have an array of definitions and schemas)
```

Comment:
Perhaps there's a more forward-looking descriptor and a more generic one than "OAMPath".

Reply:
OAMPath is to be removed anyways.

Reply:
Ok, will you open an issue to follow up on removal of this?

Reply:
Not just this, this along with a lot of other things will be stripped off. That is a part of last phase of MeshModel migration. So this will come under that legend.